src/components/Wallet/WalletSign.jsx

```diff
@@ -30,68 +29,27 @@ class WalletSign extends React.Component {
     };
   }
 
+  static getDerivedStateFromProps(props, state) {
```

Comment:
Doing this instead of the `transactionFinalized` function which was called on every re-render and was more complex than felt was necessary.